Azure/SONiC
# [sfp-refactoring] Initial support for CMIS application initialization

Signed-off-by: Dante Su <dante.su@broadcom.com>



#### Why I did it

Add initial support for CMIS application initialization and loopback

#### How I did it

1. sonic-buildimage: Migrate the Quanta IX9 to the new sfp-refactoring framework as a reference design
2. sonic-platform-common: 
2-1. sonic-xcvr/*/cmis.py: Add support for QSFP-DD application advertising and initialization
2-2. sonic-xcvr/sfp_optoe_base.py: Add support for error description and CMIS application selection
2-3. sonic_platform_base/sfp_base.py: Add stub routines for CMIS application initialization
3. sonic-platform-daemon: Add CMIS manager tasks to parallelize the per-port CMIS application initialization
4. sonic-utilities: Update sfputil and sfpshow to support CMIS application advertisement and error detection

#### How to verify it

1. Bring up Quanta IX9 with the default configuration and CMIS 4.0/5.0 QSFPDD modules
2. Issue 'show logging xcvrd' to check if the application initialization succeeded
3. Issue 'show interfaces transceiver eeprom --dom' to verify both the EEPROM and DOM information
4. Issue 'sudo sfputil show eeprom --dom' to verify both the EEPROM and DOM information
5. Issue 'sfputil show error-status -hw' to verify the error description

#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog

<table>
<tr>
<th>Repo</th>
<th>PR title</th>
<th>State</th>
</tr>
<tr>
<td>SONiC</td>
<td><a href='https://github.com/Azure/SONiC/pull/876'>[sfp-refactoring] Initial support for CMIS application initialization</a></td>
<td><img src="https://img.shields.io/github/pulls/detail/state/Azure/SONiC/876"></td>
</tr>
<tr>
<td>sonic-buildimage</td>
<td><a href='https://github.com/Azure/sonic-buildimage/pull/9157'>[sfp-refactoring] Initial support for CMIS application initialization</a></td>
<td><img src="https://img.shields.io/github/pulls/detail/state/Azure/sonic-buildimage/9157"></td>
</tr>
<tr>
<td>sonic-platform-common</td>
<td><a href='https://github.com/Azure/sonic-platform-common/pull/219'>[sfp-refactoring] Initial support for CMIS application initialization</a></td>
<td><img src="https://img.shields.io/github/pulls/detail/state/Azure/sonic-platform-common/219"></td>
</tr>
<tr>
<td>sonic-platform-daemon</td>
<td><a href='https://github.com/Azure/sonic-platform-daemons/pull/217'>[sfp-refactoring] Initial support for CMIS application initialization</a></td>
<td><img src="https://img.shields.io/github/pulls/detail/state/Azure/sonic-platform-daemons/217"></td>
</tr>
<td>sonic-utilities</td>
<td><a href='https://github.com/Azure/sonic-utilities/pull/1855'>[sfp-refactoring] Initial support for CMIS application initialization</a></td>
<td><img src="https://img.shields.io/github/pulls/detail/state/Azure/sonic-utilities/1855"></td>
</table>

#### A picture of a cute animal (not mandatory but encouraged)
